File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    List<List<Integer>> result = new ArrayList<>();
    public List<List<Integer>> levelOrder(TreeNode root) {
        traverse(root, 0);
        return result;
    }
    public void traverse(TreeNode node, int level){
        if(node == null) return;
        if(result.size() <= level){
            result.add(new ArrayList<>());
        }

        result.get(level).add(node.val);
        
        traverse(node.left, level + 1);
        traverse(node.right, level + 1);
        return;
    }
}

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
    const result = [];

    const traverse = (node, level) => {
        if(!node) return;
        if(result.length <= level){
            result.push([])
        }

        result[level].push(node.val);
        
        node.left && traverse(node.left, level + 1);
        node.right && traverse(node.right, level + 1);
    }

    traverse(root, 0)
    return result;
};